deployment/script/build-CCNP.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+set -o errexit
+
+ccnp_url=https://github.com/intel/confidential-cloud-native-primitives.git
+device_plugin=ccnp-device-plugin:0.1
+get_quote=ccnp-quote-server:0.1
+measurement=ccnp_measurement_server:0.1
+eventlog=ccnp_eventlog_server:0.1
+
+add_proxy(){
+        sed -i '/FROM .* as [b,q]/I a\ENV https_proxy='"$HTTPS_PROXY"''  "$1"
+        sed -i '/FROM .* as [b,q]/I a\ENV http_proxy='"$HTTP_PROXY"''  "$1"
+        sed -i '/FROM .* as [b,q]/I a\ENV HTTPS_PROXY='"$HTTPS_PROXY"''  "$1"
+        sed -i '/FROM .* as [b,q]/I a\ENV HTTP_PROXY='"$HTTP_PROXY"''  "$1"
+}
+
+#Download and install helm
+curl https://baltocdn.com/helm/signing.asc | sudo apt-key add -
+sudo apt-get install apt-transport-https --yes
```

Comment:
`sudo` is not recommended in shell scripts, sometimes it needs a password and the script is in non-interactive mode.

Reply:
> `sudo` is not recommended in shell scripts, sometimes it needs a password and the script is in non-interactive mode

If we need to keep "sudo" in the script, another way is to add the current user into the "sudo" group, this can be offline job done manually by the self-hosted runner admin.
In Ubuntu, usermod -aG sudo <current_user>
In RHEL, usermod -aG wheel <current_user>

If we decide to remove the "sudo", the scripts should run under root, please check if we need to have such limit.

Reply:
> I think it's better to check if helm is installed directly. And ask end-user to install it manually. Installing helm is not the scope of CCNP deployment .

Agree, please see here for reference: https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md

Reply:
> > `sudo` is not recommended in shell scripts, sometimes it needs a password and the script is in non-interactive mode
> 
> If we need to keep "sudo" in the script, another way is to add the current user into the "sudo" group, this can be offline job done manually by the self-hosted runner admin. In Ubuntu, usermod -aG sudo <current_user> In RHEL, usermod -aG wheel <current_user>
> 
> If we decide to remove the "sudo", the scripts should run under root, please check if we need to have such limit.

`sudo` is not recommended inside the shell scripts, but it can be used outside like `sudo ./build_ccnp.sh`.